addons/event/models/event.py

```diff
@@ -328,20 +365,15 @@ def copy(self, default=None):
         default = dict(default or {}, name=_("%s (copy)") % (self.name))
         return super(EventEvent, self).copy(default)
 
-    def button_draft(self):
-        self.write({'state': 'draft'})
-
-    def button_cancel(self):
-        if any('done' in event.mapped('registration_ids.state') for event in self):
-            raise UserError(_("There are already attendees who attended this event. Please reset it to draft if you want to cancel this event."))
-        self.registration_ids.write({'state': 'cancel'})
```

Comment:
I would keep the cancellation of registrations. What we could do is, in create / write of event, catch a write on stage, check if stage.pipe_cancel is True, and it this case call _action_set_cancel (private one).

Idea is: action_set_cancel would find the first cancel stage and write it as stage_id of event. _action_set_cancel implements the business meaning of cancellation, which is cancelling the registrations.